src/actions/actions.ts

```diff
@@ -180,17 +180,25 @@ export abstract class BaseMovement extends BaseAction {
       }
 
       for (let i = 0; i < count; i++) {
+          const firstIteration = (i === 0);
           const lastIteration = (i === count - 1);
           const temporaryResult = (recordedState.operator && lastIteration) ?
               await this.execActionForOperator(position, vimState) :
               await this.execAction           (position, vimState);
 
-          result = temporaryResult;
-
-          if (result instanceof Position) {
-            position = result;
-          } else if (isIMovement(result)) {
-            position = result.stop;
+          if (temporaryResult instanceof Position) {
+            result = temporaryResult;
+            position = temporaryResult;
+          } else if (isIMovement(temporaryResult)) {
+            if (firstIteration) {
+              (result as IMovement).start = temporaryResult.start;
+            }
+
+            if (lastIteration) {
+              (result as IMovement).stop = temporaryResult.stop;
+            } else {
+              position = temporaryResult.stop.getRightThroughLineBreaks();
```

Comment:
yup, it's necessary for IMovment. This is also why the logic of IMovement and Position is totally different. 

When we run a single movement without an operator, like `w`, we want to move cursor to the beginning of next word. While if we run a single movement with an operator, like `aw`, we want to move the end cursor to **the end of current word** (aka, the left-side character of the beginning of next word). Then we sucessfully get a selection of the whole word.

Above is about single movement w/ operator. Then if we run movements with an operator multple times, like `2aw`, after running the first `aw`, the end of seletion or the cursor is at the end of first word. As we need to include next following word into the selection, we need to pass the position of the begining of next word to `aw` command, then we can get the correct end position of the second `aw`. Lastly, we combine the beginning of first word and the end of last word and return the result selection.

Like I mentioned before, Movement (contains a start and a stop) is likely a selection. When executing selection with numeric prefix, I think we always need to move cursor to next character of the end of current selection.